### PR TITLE
Update stored test data

### PR DESCRIPTION
**Pull request by dstansby**
Fixes https://github.com/brainglobe/brainreg/issues/48. This includes the following improvements to tests:

- Update test data for Linux and Windows, so the tests pass.
- Save the test data from the test run to a folder in `tests/data` so if the tests fail, the new output files are accesible
- Refactor the tests so comparing each image counts as a single test. This means that **all** of the images are compared and pass/fail reported, instead of stopping at the first image that fails.
- Cache the downloaded brain atlas. Download is very slow, so caching significantly speeds up test runs.

### CODE REVIEW

**Comment by adamltyson** [comment]
Mostly looks good (and a lot faster!), but I don't understand why the output is no longer saved to a temporary directory? Seems weird to me to save the output into the test directory, and we now only keep the last run (rather than the last three).

**Comment by dstansby** [comment]
> Mostly looks good (and a lot faster!), but I don't understand why the output is no longer saved to a temporary directory? Seems weird to me to save the output into the test directory

The idea here was if the tests fail in an expected manner because of a deliberate change to the code, the output they produce is then made available to the user after running the tests instead of being cleaned up from a temporary directory.

> and we now only keep the last run (rather than the last three).

I'm a bit lost here (probably because of time off!), can you elaborate on what this means?

**Comment by adamltyson** [comment]
By default, when using `tmpdir` from `pytest`, it saves the output to a temporary directory. The last three runs of `pytest` are stored in the temporary directory before being overwritten. I find this useful to directly compare a change to the code, without having to move the output elsewhere.

**Comment by dstansby** [comment]
Ah, I didn't realise it was accesible after the tests. I've put back the temperorary directory.

File: tests/tests/test_integration/test_registration.py [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
-    "data",
-    "brain data",
-)
+test_data_dir = Path(os.getcwd()) / "tests" / "data"
 
-test_niftyreg_output = os.path.join(
-    os.getcwd(), "tests", "data", "registration_output", platform.system()
+brain_data_dir = test_data_dir / "brain data"
+expected_niftyreg_output_dir = (
+    test_data_dir / "registration_output" / platform.system()
```
Comment (paddyroddy) [inline tests/tests/test_integration/test_registration.py:17]: Just curious to know whether there would be a difference with chip architectures? `platform.system()` evaluates to `Darwin` on macOS

File: tests/tests/test_integration/test_registration.py [hunk 2]
```diff
@@ -28,12 +26,15 @@
...
+# The outputs are then tested in a separate test below
+@pytest.fixture(scope="session")
+def niftyreg_output_path(tmp_path_factory):
+    test_output_dir = tmp_path_factory.mktemp("output_dir")
     brainreg_args = [
         "brainreg",
-        data_dir,
-        output_directory,
+        str(brain_data_dir),
+        str(test_output_dir),
```
Comment (paddyroddy) [inline tests/tests/test_integration/test_registration.py:37]: I assume wherever these are going doesn't accept `Path` objects?

File: tests/tests/test_integration/test_registration.py [hunk 3]
```diff
@@ -45,48 +46,38 @@ def test_registration_niftyreg(tmpdir):
...
+        "deformation_field_0.tiff",
+        "deformation_field_1.tiff",
+        "deformation_field_2.tiff",
+        "downsampled.tiff",
+        "downsampled_brain data.tiff",
+        "downsampled_standard.tiff",
+        "downsampled_standard_brain data.tiff",
+        "registered_atlas.tiff",
+        "registered_hemispheres.tiff",
+    ],
```
Comment (paddyroddy) [inline tests/tests/test_integration/test_registration.py:70]: Kind of pointless but could do
```suggestion
    [
        "boundaries.tiff",
        "downsampled.tiff",
        "downsampled_brain data.tiff",
        "downsampled_standard.tiff",
        "downsampled_standard_brain data.tiff",
        "registered_atlas.tiff",
        "registered_hemispheres.tiff",
    ] + [f"deformation_field_{i}.tiff" for i in range(3)]
```

**Review by paddyroddy — APPROVED** [review]

File: tests/tests/test_integration/test_registration.py (hunk 2 above)
Comment (dstansby) [reply tests/tests/test_integration/test_registration.py:37]: Yep

**Pull request merged**